src/common/bit_cast.hpp

```diff
@@ -37,8 +37,8 @@ inline T bit_cast(const U &u) {
     static_assert(sizeof(T) == sizeof(U), "Bit-casting must preserve size.");
     // Use std::is_pod as older GNU versions do not support
     // std::is_trivially_copyable.
-    static_assert(std::is_pod<T>::value, "T must be trivially copyable.");
-    static_assert(std::is_pod<U>::value, "U must be trivially copyable.");
+    static_assert(std::is_trivially_copyable<T>::value, "T must be trivially copyable.");
```

Comment:
This will cause build errors when an old `libstdc++` is used since it doesn't support `std::is_trivially copyable` trait.
This will affect GNU 4.8 and Clang that uses the old `libstdc++`.

Reply:
Understood. But I've done all I was going to. I've reported the issue and I've submitted a fix.

Supporting older compilers costs more. So I'll leave it to those who have an interest. I don't.

Reply:
oneDNN supports a variety of compilers. We cannot accept a fix that introduces a regression.

Reply:
I understand this won't be accepted. I'm saying I've fixed my problem and won't work on this any more.

But you still have an open bug to fix.

Reply:
Maybe just change `is_pod` to `is_trivial`, it should be sufficient to check whether the given type can be used in `memcpy`?

Reply:
That's a superset: a trivial type is trivially copyable and trivially destructible. So, yes.

Is `is_trivial` available in GCC 4.8?

Reply:
I think so

Reply:
Updating.